add custom easyconfig parameter `with_cxx_backtrace` to GCC easyblock to enable the build of the libstdcxx-backtrace library

maxim-masterov opened the pull request:
Add an option to build a static library for the stack trace support in C++23.

maxim-masterov committed:
Add a flag to enable stacktrace

maxim-masterov committed:
Remove trailing whitespace

Thyre commented:
@boegelbot please test @ jsc-zen3
EB_ARGS="GCCcore-13.3.0.eb GCCcore-14.2.0.eb GCCcore-14.3.0.eb GCCcore-15.2.0.eb --installpath=/tmp/$USER/ebpr-3538 --try-amend=with_cxx_backtrace=True"

Thyre commented on easybuild/easyblocks/g/gcc.py:
```diff
@@ -643,6 +644,10 @@ def configure_step(self):
+        # enable builind of libstdc++_libbacktrace
+        if self.cfg['with_cxx_backtrace']:
+            self.configopts += " --enable-libstdcxx-backtrace"
+
```
With the current approach, having `with_cxx_backtrace=False` by default, we should probably also disable support via a flag for GCC versions supporting backtrace.

We could go the route and define `with_cxx_backtrace` as `None` by default, passing no flag and just use what GCC has by default. Users could then pass `True` to explicitly enable support, and `False` to explicitly disable support.

maxim-masterov committed:
Modify 'with_cxx_backtrace' option handling in gcc.py

Change 'with_cxx_backtrace' default value to None and update condition for enabling libstdc++_libbacktrace.

maxim-masterov committed:
Fix comment formatting in gcc.py
